app/Repositories/Eloquent/BaseRepository.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+
+namespace App\Repositories\Eloquent;
+
+
+use App\Repositories\Interfaces\RepositoryInterface;
+
+abstract class BaseRepository implements RepositoryInterface
```

Comment:
Repository dùng làm gì e

Reply:
* Repository Pattern là lớp trung gian giữa tầng Business Logic và Data Access
* Thay vì viết code truy xuất, thao tác với Database, Model trong Controller thì ta viết ở đây để giảm sự phình to của Controller. Ngoài ra giảm sự phụ thuộc giữa các Class, tăng khả năng bảo trì mở rộng sau này, tránh việc lặp code.
* Muốn sử dụng thì inject vào thông qua __construct (kỹ thuật Dependency injection nhờ service container trong laravel).
* Tạo ra 1 class chung đặt tên là EloquentRepository để xử lý tác vụ crud chung. Sau đó, chúng ta sẽ tạo ra các class con extend class EloquentRepository và truyền vào models tương ứng để xử lý.

Reply:
Trả về null ạ